packages/cinterop/src/commonMain/kotlin/io/realm/interop/NetworkTransport.kt

```diff
@@ -0,0 +1,43 @@
+package io.realm.interop
+
+interface NetworkTransport {
+
+    companion object {
+        const val GET = "get"
+        const val POST = "post"
+        const val PATCH = "patch"
+        const val PUT = "put"
+        const val DELETE = "delete"
+    }
+
+    val authorizationHeaderName: String?
+    val customHeaders: Map<String, String>
+
+    fun sendRequest(
+        method: String,
+        url: String,
+        headers: Map<String, String>,
+        body: String,
+        usesRefreshToken: Boolean
+    ): Response
+}
+
+data class Response(
+    val httpResponseCode: Int,
+    val customResponseCode: Int,
+    val headers: Map<String, String>,
+    val body: String
+) {
+    // Returns the HTTP headers in a JNI friendly way where it is being serialized to a
+    // String array consisting of pairs of { key , value } pairs.
+    fun getJNIFriendlyHeaders(): Array<String?> {
+        val jniHeaders = arrayOfNulls<String>(headers.size * 2)
+        var i = 0
+        for ((key, value) in headers) {
```

Comment:
This could be done using 
```Kotlin
return headers.flatMap { listOf(it.key) + it.value }

Reply:
Less lines of code, bu it feels quite inefficient (though I don't know if it is optimized in some way under the hood), so will just keep the existing code.